stdlib/public/core/EmbeddedPrint.swift

```diff
@@ -0,0 +1,96 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftShims
+
+// TODO: This is all a stop-gap so that at least some types are printable in
+// embedded Swift, in an embedded-programming friendly way (we mainly need
+// printing to not need to heap allocate).
+
+@_silgen_name("putchar")
+@discardableResult
+func putchar(_: CInt) -> CInt
+
+public func print(_ string: StaticString, terminator: StaticString = "\n") {
+  var p = string.utf8Start
+  while p.pointee != 0 {
+    putchar(CInt(p.pointee))
```

Comment:
does the semantics of all `putchar` implementations always support proper UTF8?

Reply:
This is one of the reasons why we should call this solution a stop-gap and aim to replace it soon -- there is no guarantee that Unicode is handled properly via putchar.

However, I tend to think that it's a reasonable starting point, because: (1) putchar sends *bytes*, it's oblivious to the encoding (2) in practice, in embedded use cases for debug-logging purposes, the other end of this stream is most commonly going to be a terminal or a file on a connected host's main OS, and the most reasonably expected encoding by terminals and file viewers is probably UTF-8. In which case this solution (sending the UTF-8 bytes from the string via putchar) is actually going to work.